src/scripts/components/widgets/modals/Modal.js

```diff
@@ -71,17 +73,26 @@ const Content = styled.div`
 
 class Modal extends Component<Props, void> {
   render () {
+    const isVisible = this.props.showModal
+    document.onkeydown = event => {
+      if (event.keyCode === 27) {
+        this.props.closeModal()
+      }
+    }
+
     return (
-      <Wrapper show={this.props.show}>
-        <Container show={this.props.show}>
+      <Wrapper show={isVisible}>
+        <Container show={isVisible}>
           <CloseButton onClick={this.props.closeModal}>
             <SVG>
               <use xlinkHref="#icon-close" />
             </SVG>
           </CloseButton>
-          <Content>{this.props.children}</Content>
+          <Content>
+            {this.props.modalContent === 'ModalStake' && <ModalStake />}
```

Comment:
This is fine for now since we have just one modal (I think), but it might be good to move this to a `renderModal` method.

```javascript
renderModal() {
  const { modalContent } = this.props
  switch(modalContent) {
    case 'ModalStake':
      return <ModalStake />
  }
}
```

Then just call this method in `render`.

Also, I think it would be very good to not use raw strings for the modals. Instead, create a `constants/ModalConstants.js` file and define them as exportable constants